### ISSUE TITLE
Improve `Worker()` API

### ISSUE THREAD

**Issue by nvie** [nvie/rq#255]
See https://github.com/nvie/rq/issues/247#issuecomment-23927849

Currently, Worker takes a list of Queue instances. I think there's two things we need to improve here:

* Expect generators, so this stuff works in Python 3 just as naturally:
  
  ```python
queue_names = ['high', 'normal', 'default']
w = Worker(list(map(Queue, queue_names))) # extra list() wrap should be unnecessary
  ```

* Why even bother doing the Queue() wrapping yourself? I think the Worker API should allow for 
  
  ```python
queue_names = ['high', 'normal', 'default']
w = Worker(queue_names)  # Worker() should accept (string) names
  ```

**Comment by aleiphoenix** [comment]
+1 for this, and the documentation's still not updated.

**Comment by RyanMTB** [comment]
I found the worker API a little unintuitve while I was learning it.  Specifically ran into an error with this because i forgot the list

```python
queue_names = ['high', 'normal', 'default']
w = Worker(map(Queue, queue_names)) #error
w = Worker(list(map(Queue, queue_names)))  #noerror
````

See my pull request above for a fix.  I would love some feedback on it.  I am new to contributing to open source.

**Comment by selwin** [comment]
Fixed in #531
